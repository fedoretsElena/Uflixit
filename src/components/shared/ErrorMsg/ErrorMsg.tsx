import React from "react";

interface IProps {
    msg: string | null;
}

const ErrorMsg = (props: IProps) => (
    <div className="alert alert-warning" role="alert">
        {props.msg}
    </div>
);

export default ErrorMsg;
