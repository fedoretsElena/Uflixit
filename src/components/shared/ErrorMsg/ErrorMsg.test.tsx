import ErrorMsg from "./ErrorMsg";

describe("ErrorMsg component", () => {
   const msg = 'Waning msg';

    it("should render correctly", () => {
       const component = shallow(<ErrorMsg msg={msg} />);

       expect(component).toMatchSnapshot();
    });

    it("should show particular text from props", () => {
       const component = mount(<ErrorMsg msg={msg}/>);

       const text = component.find('.alert')
           .text();

       expect(text).toEqual(msg);
    });
});