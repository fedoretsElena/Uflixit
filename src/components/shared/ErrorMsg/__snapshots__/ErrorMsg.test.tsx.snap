// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMsg component should render correctly 1`] = `
<div
  className="alert alert-warning"
  role="alert"
>
  Waning msg
</div>
`;
